test/functional/rpc_psbt.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+# Copyright (c) 2014-2016 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test the Partially Signed Transaction RPCs.
+"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+
+# Create one-input, one-output, no-fee transaction:
+class PSBTTest(BitcoinTestFramework):
+
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 3
+
+    def run_test(self):
+
+        # Get some money and activate segwit
+        self.nodes[0].generate(500)
+
+        # Create and fund a raw tx for sending 10 BTC
+        psbtx1 = self.nodes[0].walletcreatefundedpsbt([], {self.nodes[2].getnewaddress():10})['psbt']
+
+        # Node 1 should not be able to add anything to it but still return the psbtx same as before
+        psbtx = self.nodes[1].walletprocesspsbt(psbtx1)['psbt']
+        assert_equal(psbtx1, psbtx)
+
+        # Sign the transaction and send
+        signed_tx = self.nodes[0].walletprocesspsbt(psbtx)['psbt']
+        final_tx = self.nodes[0].finalizepsbt(signed_tx)['hex']
+        self.nodes[0].sendrawtransaction(final_tx)
+
+        # Create p2sh, p2wpkh, and p2wsh addresses
+        pubkey0 = self.nodes[0].getaddressinfo(self.nodes[0].getnewaddress())['pubkey']
+        pubkey1 = self.nodes[1].getaddressinfo(self.nodes[1].getnewaddress())['pubkey']
+        pubkey2 = self.nodes[2].getaddressinfo(self.nodes[2].getnewaddress())['pubkey']
+        p2sh = self.nodes[1].addmultisigaddress(2, [pubkey0, pubkey1, pubkey2], "", "legacy")['address']
+        p2wsh = self.nodes[1].addmultisigaddress(2, [pubkey0, pubkey1, pubkey2], "", "bech32")['address']
+        p2wpkh = self.nodes[1].getnewaddress("", "legacy")
```

Comment:
That doesn't look very witnessy.

Reply:
Fixed. Also added tests for p2pkh and p2sh nested things.